Não esta auto preenchendo campos de autoridades nem de vocabulário

Testado nas versões
Biblivre 5 (versão da instalação do site http://biblivre.org.br/) **Aqui funciona**
branch Issue-330 **Aqui tem o problema** (ultima versão que peguei)

**Problema**
Após cadastrar um registro em autoridade e outro em vocabulário, ao cadastrar uma obra não esta auto preenchendo os campos cadastrados de autoridade e vocabulário.

Como reproduzir (Autoridades):
- Acessar "Catalogação" -> "Autoridades" ;
- Clicar em "Novo registro";
- Em "Tipo de autor" selecionar "Pessoa";
- Em "Autor - Nome pessoal (100)"  -> "Sobrenome e/ou prenome do autor" digitar alguma coisa(ex: testando autor);
- Salvar a modificação;
- Ir para "Catalogação" -> "Bibliográfica";
- Clicar em "Novo registro";
- Ir até o campo "Autor - Nome pessoal (100)" e começar a digitar o que foi cadastrado em "Autoridades" (ex: testan);
- Verificar que não auto completou.

Como reproduzir (Vocabulário):
- Acessar "Catalogação" -> "Vocabulário" ;
- Clicar em "Novo registro";
- Em "TE(150)" -> "Termo tópico adotado" digitar alguma coisa(ex: testando assunto);
- Salvar a modificação;
- Ir para "Catalogação" -> "Bibliográfica";
- Clicar em "Novo registro";
- Ir até o campo "Assunto - Tópico(650)" e começar a digitar o que foi cadastrado em "Vocabulário" (ex: testan);
- Verificar que não auto completou.

Opa!
Parece que tem algum problema ao restaurar backup, percebi agora usando a branch issue-338. 
Mas **não** parece ser dela o problema. Peguei a master e tbm da erro ao restaurar backup.
Na brach issue-330 o erro não ocorre usando o mesmo arquivo de backup.

A seguir tem o log e lá no final coloquei o arquivo de backup(tem apenas 1 livro cadastrado e com poucos campos) caso queira testar.
Se preferir que crie uma nova issue só avisar! Abraço
```
[Log de restauração de backup do BIBLIVRE 5]
single_schema => single_schema
Restoring 2 schemas for a total of 15943 SQL lines
Starting psql
Renaming schema single to _single_1657675063716
Renaming schema global to _global_1657675063716
Processing schema for 'global'
Processing data for 'global'
biblivre.administration.backup.exception.RestoreException: java.io.IOException: Broken pipe
	at biblivre.administration.backup.RestoreBO._writeLine(RestoreBO.java:856)
	at biblivre.administration.backup.RestoreBO.lambda$processRestore$0(RestoreBO.java:201)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.nio.file.FileChannelLinesSpliterator.forEachRemaining(FileChannelLinesSpliterator.java:133)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at biblivre.administration.backup.RestoreBO.processRestore(RestoreBO.java:199)
	at biblivre.administration.backup.RestoreBO._processGlobalSchema(RestoreBO.java:310)
	at biblivre.administ
[backupTesteMaster.zip](https://github.com/cleydyr/biblivre/files/9098063/backupTesteMaster.zip)
ration.backup.RestoreBO.restoreBackup(RestoreBO.java:246)
	at biblivre.administration.backup.RestoreBO.restore(RestoreBO.java:135)
	at biblivre.administration.setup.Handler.restore(Handler.java:388)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at biblivre.core.controllers.Controller.processRequest(Controller.java:181)
	at biblivre.core.controllers.SchemaServlet.processDynamicRequest(SchemaServlet.java:177)
	at biblivre.core.controllers.SchemaServlet.processDynamicRequest(SchemaServlet.java:138)
	at biblivre.core.controllers.SchemaServlet.doPost(SchemaServlet.java:133)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at biblivre.core.controllers.ExtendedRequestResponseFilter.doFilter(ExtendedRequestResponseFilter.java:63)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:890)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1787)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:833)
	Suppressed: java.io.IOException: Stream closed
		at java.base/java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:445)
		at java.base/java.io.OutputStream.write(OutputStream.java:162)
		at java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:123)
		at java.base/sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:234)
		at java.base/sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:304)
		at java.base/sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:282)
		at java.base/sun.nio.cs.StreamEncoder.write(StreamEncoder.java:132)
		at java.base/java.io.OutputStreamWriter.write(OutputStreamWriter.java:205)
		at java.base/java.io.BufferedWriter.flushBuffer(BufferedWriter.java:120)
		at java.base/java.io.BufferedWriter.close(BufferedWriter.java:268)
		at biblivre.administration.backup.RestoreBO.restoreBackup(RestoreBO.java:237)
		... 37 more
Caused by: java.io.IOException: Broken pipe
	at java.base/java.io.FileOutputStream.writeBytes(Native Method)
	at java.base/java.io.FileOutputStream.write(FileOutputStream.java:349)
	at java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:123)
	at java.base/sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:234)
	at java.base/sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:304)
	at java.base/sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:282)
	at java.base/sun.nio.cs.StreamEncoder.write(StreamEncoder.java:132)
	at java.base/java.io.OutputStreamWriter.write(OutputStreamWriter.java:205)
	at java.base/java.io.BufferedWriter.flushBuffer(BufferedWriter.java:120)
	at java.base/java.io.BufferedWriter.write(BufferedWriter.java:233)
	at java.base/java.io.Writer.write(Writer.java:249)
	at biblivre.administration.backup.RestoreBO._writeLine(RestoreBO.java:853)
	... 52 more
```
[backupTesteMaster.zip](https://github.com/cleydyr/biblivre/files/9098066/backupTesteMaster.zip)

--- a/src/main/java/biblivre/cataloging/PaginableCatalogingHandler.java
+++ b/src/main/java/biblivre/cataloging/PaginableCatalogingHandler.java
@@ -36,7 +36,9 @@ import biblivre.core.file.DiskFile;
 import biblivre.marc.MaterialType;
 import biblivre.marc.RecordStatus;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.UUID;
 import java.util.stream.Collectors;
@@ -47,6 +49,8 @@ import org.marc4j.marc.Record;
 
 public abstract class PaginableCatalogingHandler extends CatalogingHandler {
     protected PaginableRecordBO paginableRecordBO;
+
+    protected Map<String, PaginableRecordBO> recordBOs = new HashMap<>();
 
     public void search(ExtendedRequest request, ExtendedResponse response) {
         String searchParameters = request.getString("search_parameters");
@@ -254,8 +258,10 @@ public abstract class PaginableCatalogingHandler extends CatalogingHandler {
             case BIBLIO:
             case AUTHORITIES:
             case VOCABULARY:
+                PaginableRecordBO autocompleteRecordBO = recordBOs.get(type.name());
+
                 DTOCollection<AutocompleteDTO> autocompletion =
-                        type.getAutocompletion(paginableRecordBO, query);
+                        type.getAutocompletion(autocompleteRecordBO, query);
 
                 this.json.putOpt("data", autocompletion.toJSONObject());
 
@@ -308,5 +314,9 @@ public abstract class PaginableCatalogingHandler extends CatalogingHandler {
 
     public void setDefaultMaterialType(MaterialType defaultMaterialType) {
         this.defaultMaterialType = defaultMaterialType;
+    }
+
+    public void setRecordBOs(Map<String, PaginableRecordBO> recordBOs) {
+        this.recordBOs = recordBOs;
     }
 }

--- a/src/main/java/biblivre/cataloging/holding/HoldingDAOImpl.java
+++ b/src/main/java/biblivre/cataloging/holding/HoldingDAOImpl.java
@@ -57,6 +57,8 @@ import org.marc4j.marc.Record;
 
 public class HoldingDAOImpl extends RecordDAOImpl implements HoldingDAO {
 
+    private static final int ADMIN_OR_REMOVED_LOGIN_ID = 0;
+
     public static HoldingDAOImpl getInstance() {
         return AbstractDAO.getInstance(HoldingDAOImpl.class);
     }
@@ -287,7 +289,7 @@ public class HoldingDAOImpl extends RecordDAOImpl implements HoldingDAO {
 
             if (ldto != null) {
                 pst.setString(2, ldto.getLogin());
-                pst.setInt(3, 0);
+                pst.setInt(3, ADMIN_OR_REMOVED_LOGIN_ID);
             } else {
                 pst.setNull(2, java.sql.Types.VARCHAR);
                 pst.setNull(3, java.sql.Types.INTEGER);

--- a/src/main/resources/applicationContext.xml
+++ b/src/main/resources/applicationContext.xml
@@ -100,16 +100,19 @@
 		class="biblivre.cataloging.authorities.Handler" scope="prototype">
 		<property name="paginableRecordBO" ref="authorityRecordBO" />
 		<property name="reportsBO" ref="reportsBO" />
+		<property name="recordBOs" ref="recordBOs" />
 	</bean>
 
 	<bean id="bibliographicHandler" name="bibliographicHandler"
 		class="biblivre.cataloging.bibliographic.Handler" scope="prototype">
 		<property name="paginableRecordBO" ref="biblioRecordBO" />
+		<property name="recordBOs" ref="recordBOs" />
 	</bean>
 
 	<bean id="vocabularyHandler" name="vocabularyHandler"
 		class="biblivre.cataloging.vocabulary.Handler" scope="prototype">
 		<property name="paginableRecordBO" ref="vocabularyRecordBO" />
+		<property name="recordBOs" ref="recordBOs" />
 	</bean>
 
 	<bean id="holdingHandler" name="holdingHandler"

